### ISSUE TITLE
Recent commit messages are always empty

### ISSUE THREAD

**Issue by kcmr** [ForkIssues/Tracker#52]
Recent commit messages are very useful when you reset HEAD to a previous commit and want to rewrite some discarded commits, etc., but I haven't been able to use it never because they are always empty or unavailable.

**Comment by DanPristupov** [comment]
Right... If I remember correctly, Fork creates the list of recent commits like following `git log -n 5 HEAD`. Since you moved the HEAD backward, there's no way to get the messages in front of HEAD (i.e. the squashed ones).

I'll think about this during the weekend, but it looks like there's nothing I can do to "fix" this.

But there's a better way to do achieve what you want. I'd recommend you to use interactive rebase instead. There's a corresponding topic already: https://github.com/ForkIssues/Tracker/issues/41.
TLDR: instead of reset, RI your commits to previous one and `squash` them.

P.S. RI is also able to make backup of the initial state which is very handy when user rewrites history.

**Comment by kcmr** [comment]
Thank you. 
I had no idea that that feature exists! :)
